Capstone/API/APIManager.m

```diff
@@ -61,26 +64,123 @@ - (void)logout {
         UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
         WelcomeViewController *welcomeViewController = [storyboard instantiateViewControllerWithIdentifier:@"WelcomeView"];
         mySceneDelegate.window.rootViewController = welcomeViewController;
+        [[CacheManager sharedManager] didlogout];
     }];
 }
 
 - (void)fetchMapDataWithCompletion:(NSArray *)coordinates completion:(void(^)(NSArray *posts, NSError *error))completion {
+    NSDate *today = [NSDate date];
+    [[NSCalendar currentCalendar] rangeOfUnit:NSCalendarUnitDay startDate:&today interval:NULL forDate:today];
     PFQuery *query = [PFQuery queryWithClassName:@"Post"];
     [query orderByDescending:@"createdAt"];
+    [query whereKey:@"createdAt" greaterThanOrEqualTo:today];
     [query whereKey:@"Location" withinPolygon:coordinates];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *parsePosts, NSError *error) {
+        if (parsePosts != nil) {
+            completion(parsePosts, error);
+        }
+    }];
+}
 
-    // fetch data asynchronously
+- (void)fetchCalendarDataWithCompletion:(PFUser *)user date:(NSDate *)date completion:(void(^)(NSArray *posts, NSError *error))completion {
+    NSCalendar* calendar = [NSCalendar currentCalendar];
+    unsigned unitFlags = NSCalendarUnitYear | NSCalendarUnitMonth |  NSCalendarUnitDay;
+    NSDateComponents* comps = [calendar components:unitFlags fromDate:date];
+    [comps setMonth:[comps month]+1];
+    [comps setDay:0];
+    NSDate *lastDateMonth = [calendar dateFromComponents:comps];
+    lastDateMonth = [lastDateMonth dateByAddingTimeInterval:(60*60*24)];
+    [comps setMonth:[comps month]-1];
+    [comps setDay:1];
+    NSDate *firstDateMonth = [calendar dateFromComponents:comps];
+    [[NSCalendar currentCalendar] rangeOfUnit:NSCalendarUnitDay startDate:&firstDateMonth interval:NULL forDate:firstDateMonth];
+    PFQuery *query = [PFQuery queryWithClassName:@"Post"];
+    [query whereKey:@"UserID" equalTo:user.username];
+    [query whereKey:@"createdAt" greaterThanOrEqualTo:firstDateMonth];
+    [query whereKey:@"createdAt" lessThan:lastDateMonth];
     [query findObjectsInBackgroundWithBlock:^(NSArray *parsePosts, NSError *error) {
         if (parsePosts != nil) {
             completion(parsePosts, error);
         }
     }];
 }
 
+// called after user posts to fetch new post into cache
+- (void)fetchTodayCalendarDataWithCompletion:(PFUser *)user completion:(void(^)(Post *_Nullable post, BOOL success))completion {
```

Comment:
Let's change this function name to be clear that this is only catching the last post to add it to cache.